src/generic/gridctrl.cpp

```diff
@@ -157,6 +157,13 @@ using namespace wxGridPrivate;
 
 // Enables a grid cell to display a formatted date
 
+wxGridCellDateRenderer::wxGridCellDateRenderer(const wxGridCellDateRenderer& other)
+    : m_oformat(other.m_oformat)
+    , m_tz(other.m_tz)
+{
+    SetClientDataContainer(other.GetClientDataContainer());
```

Comment:
I think it's really, really annoying to have to explicitly do this everywhere in the code. It would be better if it could be done once in the base class ctor instead, so I think we should add a (protected) copy ctor `wxGridCellRenderer(const wxGridCellRenderer& other)` and do it there.

The only problem with this is that if we do it, we suppress the generation of the default ctor for `wxGridCellRenderer` and we do need to have it to avoid breaking the existing code relying on it, if nothing else. So we're going to add it too. But perhaps we should make it inaccessible while building wxWidgets itself, i.e. put `#ifndef WXBUILDING` around it, so that it would be impossible to forget to use the copy ctor in wx own code.

However even without this hack, having a copy (and default) ctor would be better than having to manually call `SetClientDataContainer()` everywhere. Could you please do it like this?

Reply:
Sorry, I might be misunderstanding the question, but I was only speaking about avoid the need for all these explicit `SetClientDataContainer()` calls by doing it in the copy ctor. It wouldn't change anything for creating new instances, AFAICS. What problem exactly do you have in mind?

Reply:
The problem is, as you mentions, deleted default ctors. I could spread copy ctors around, but I also think the default ctors need to be there. Btw, what is expected of compilers for wxWidgets - is "= default;" (C++11) allowed in the code?

Reply:
Unfortunately we still support pre-C++11 compilers such as MSVS 2008 in 3.2 (which is why we need to release it a.s.a.p. in order to be able to drop support for them).

I think for now you should just add default ctors, that we need to preserve for compatibility, and disregard my hack with `WXBUILDING`, it's probably not worth the complications.

Thanks!